SocketRocket/SRWebSocket.m

```diff
@@ -527,6 +528,12 @@ - (void)didConnect
     _secKey = [keyBytes SR_stringByBase64Encoding];
     assert([_secKey length] == 24);
     
+    // set header for http basic auth
+    if(_url.user && _url.password) {
```

Comment:
Further, to be super-defensive, should be `if (_url.user.length && _url.password.length)` to protect against empty strings.